lms/djangoapps/teams/tests/test_views.py

```diff
@@ -1824,6 +2332,62 @@ def test_student_leave_instructor_managed_team(self, *args):  # pylint: disable=
         self.delete_membership(
             self.solar_team.team_id, self.users['student_enrolled'].username, 403, user='student_enrolled')
 
+    @ddt.unpack
+    @ddt.data(
+        ('student_enrolled', 'student_masters', 404),
+        ('student_enrolled', 'student_enrolled', 403),
+        ('student_masters_not_on_team', 'student_masters', 404),
+        ('student_masters_not_on_team', 'student_masters_not_on_team', 404),
+        ('student_masters', 'student_masters', 204),
+        ('staff', 'student_masters', 204),
+        ('staff', 'staff', 404),
+    )
+    def test_organization_protection_status(self, user, user_to_remove, expected_status):
+        """
+        As different users, attempt to remove themselves or studet_masters from masters_only team.
+        (only student_masters is actually on this team)
+        Only staff and the student_masters should be able to remove, and users outside of the
+        organization_protected bubble should not be able to tell that the team exists in
+        any way.
+
+        TODO: student_enrolled should not be able to tell that masters_only team exists, he should
+        get a 404 on both calls
+        """
+        self.delete_membership(
+            self.masters_only_team.team_id,
+            self.users[user_to_remove].username,
+            expected_status,
+            user=user
+        )
+
+    @ddt.unpack
+    @ddt.data(
+        ('student_enrolled', 'student_on_team_1_private_set_1', 404),
+        ('student_enrolled', 'student_enrolled', 403),
+        ('student_on_team_1_private_set_1', 'student_on_team_1_private_set_1', 403),
+        ('student_on_team_2_private_set_1', 'student_on_team_1_private_set_1', 404),
+        ('student_on_team_2_private_set_1', 'student_on_team_2_private_set_1', 403),
+        ('staff', 'student_on_team_1_private_set_1', 204),
+        ('staff', 'staff', 404),
+    )
+    def test_teamset_type(self, user, user_to_remove, expected_status):
```

Comment:
nit: What about calling this test_removal_ability_due_to_membership or some other name that indicates we are testing removal?